src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs

```diff
@@ -50,5 +57,32 @@ public override bool IsLazyEvalParam(int index)
         {
             return LamdaParamMask.TestBit(index);
         }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+
+            var isValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+
+            // Check if all record names of args exist against arg types and if its possible to coerce.
+            if (isValid)
+            {
+                for (var i = 0; i < args.Length; i++)
+                {
+                    DType curType = argTypes[i];
+
+                    if (curType.IsRecord && !curType.CheckAggregateNames(this.ParamTypes[i], args[i], errors, SupportsParamCoercion, context.Features.PowerFxV1CompatibilityRules))
```

Comment:
>SupportsParamCoercion [](http://example.com/codeflow?start=110&length=21)

Some functions support coercion only for certain parameters, so we should call `SupportCoercionForArg(i)` instead

Reply:
updated